components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -1007,6 +1007,10 @@ public static void startTenantFlow(String tenantDomain) throws IdentityApplicati
 
         String userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (MultitenantConstants.SUPER_TENANT_ID != tenantId && !IdentityTenantUtil.getTenant(tenantId).isActive()) {
+            throw new IdentityApplicationManagementException("Error when setting tenant domain. "
```

Comment:
This should a client Exception. The messaging needs to change. `Error when setting tenant domain` this is wrong

Reply:
```
"throw new IdentityApplicationManagementException("Cannot start the tenant flow. "
                    + tenantDomain + " is deactivated.");
```
WDYT?

Reply:
yeah, and it should be IdentityApplicationManagementClinetException

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/4361/commits/cc2d053fa557680f301bf9a3ce064c832b5fbc1b